cpp/src/arrow/dataset/file_parquet.h

```diff
@@ -46,6 +47,23 @@ class ARROW_DS_EXPORT ParquetWriteOptions : public FileWriteOptions {
 /// \brief A FileFormat implementation that reads from Parquet files
 class ARROW_DS_EXPORT ParquetFileFormat : public FileFormat {
  public:
+  /// \defgroup parquet-file-format-reader-properties properties which correspond to
+  /// members of parquet::ReaderProperties.
+  ///
+  /// @{
+  bool use_buffered_stream = parquet::DEFAULT_USE_BUFFERED_STREAM;
+  int64_t buffer_size = parquet::DEFAULT_BUFFER_SIZE;
+  std::shared_ptr<parquet::FileDecryptionProperties> file_decryption_properties;
+  /// @}
+
+  /// \defgroup parquet-file-format-arrow-reader-properties properties which correspond to
+  /// members of parquet::ArrowReaderProperties.
+  ///
+  /// @{
+  std::unordered_set<int> read_dict_indices;
```

Comment:
This should probably be a set of field names rather than field indices; otherwise we are essentially requiring that the schema be known as the format is being created